expression/builtin_time_vec.go

```diff
@@ -1029,11 +1029,37 @@ func (b *builtinSubDateAndDurationSig) vecEvalString(input *chunk.Chunk, result
 }
 
 func (b *builtinDayOfYearSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinDayOfYearSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETDatetime, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err = b.args[0].VecEvalTime(b.ctx, input, buf); err != nil {
+		return err
```

Comment:
Should we also check `handleInvalidTimeError` here?

Reply:
Yes, I've changed it.